pkg/controller/constraint/constraint_controller.go

```diff
@@ -456,3 +613,46 @@ func (c *ConstraintsCache) reportTotalConstraints(ctx context.Context, reporter
 		}
 	}
 }
+
+func ShouldGenerateVap(useVapLabel string) bool {
+	if VapEnforcement == VapFlagGatekeeperDefault {
+		return useVapLabel == "yes"
+	}
+	if VapEnforcement == VapFlagVapDefault {
+		return useVapLabel != "no"
+	}
+	return false
+}
+
+func IsVapAPIEnabled() bool {
+	vapMux.Lock()
+	defer vapMux.Unlock()
+
+	if VapAPIEnabled != nil {
+		return *VapAPIEnabled
+	}
+	groupVersion := schema.GroupVersion{Group: "admissionregistration.k8s.io", Version: "v1beta1"}
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Info("IsVapAPIEnabled InClusterConfig", "error", err)
+		VapAPIEnabled = new(bool)
+		*VapAPIEnabled = false
+		return false
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Info("IsVapAPIEnabled NewForConfig", "error", err)
+		*VapAPIEnabled = false
+		return false
+	}
+	if _, err := clientset.Discovery().ServerResourcesForGroupVersion(groupVersion.String()); err != nil {
+		log.Info("IsVapAPIEnabled ServerResourcesForGroupVersion", "error", err)
```

Comment:
should this be log.Error?

Reply:
other places too. or are you using this for debugging?

Reply:
we dont want to return any errors, just default to false.